### PR TITLE
fix: Add Basic Functionality to Settings Search

### LINKED ISSUE #25441: [Bug]: Settings - `Basic Functionality` does not appear on Settings search
### Describe the bug

`Basic Functionality` does not appear on Settings search

### Expected behavior

_No response_

### Screenshots/Recordings

Screenshot from 2024-06-20 15-59-07

### Steps to reproduce

1. Go to settings
2. Look for basic functionality
3. See there are no results

### Error messages or log output

_No response_

### Version

11.16.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Adds "Basic Functionality" to the Settings search

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25185?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/25441

## **Manual testing steps**

1. Go to Settings
2. Search the term "Basic"
3. See "Basic Functionality" and click it
4. End up on top of the Privacy tab

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/helpers/constants/settings.js [hunk 1]
```diff
@@ -151,46 +151,54 @@ const SETTINGS_CONSTANTS = [
...
   },
   // securityAndPrivacy settingsRefs[0]
+  {
+    tabMessage: (t) => t('securityAndPrivacy'),
+    sectionMessage: (t) => t('basicConfigurationLabel'),
+    descriptionMessage: (t) => t('basicConfigurationDescription'),
+    route: `${SECURITY_ROUTE}#basic-functionality-toggle`,
+    icon: 'fa fa-lock',
+  },
+  // securityAndPrivacy settingsRefs[1]
```
Comment (bergeron) [inline ui/helpers/constants/settings.js:161]: This PR did allow me to search + jump to the basic toggle.  But I notice some other settings breaking.  Looking for fix

Before on latest develop, IPFS gateway would jump correctly:

https://github.com/MetaMask/metamask-extension/assets/3500406/1d9b6b9b-e880-459c-aa41-6757b662f6ce

But on this branch it jumps me to autodetect instead:


https://github.com/MetaMask/metamask-extension/assets/3500406/40240b7e-6977-4191-877e-d3555ea8df5f

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 2]
```diff
@@ -955,7 +955,7 @@ export default class SecurityTab extends PureComponent {
 
     return (
       <Box
-        ref={this.settingsRefs[3]}
+        ref={this.settingsRefs[0]}
         className="settings-page__content-row"
         display={Display.Flex}
         flexDirection={FlexDirection.Row}
```
Comment (bergeron) [inline ui/pages/settings/security-tab/security-tab.component.js:961]: Totally unrelated but 3 lines down seems like the wrong `data-testid`

File: ui/helpers/constants/settings.js (hunk 1 above)
Comment (bergeron) [reply ui/helpers/constants/settings.js:161]: Ok I'm more confused, since I seem to be able to search for basic functionality even on latest develop

Comment (bergeron) [reply ui/helpers/constants/settings.js:161]: I think this is the fix.  Just shifting the other indices that incremented (all except 18)

https://github.com/MetaMask/metamask-extension/commit/1dc663df37ce0c4cdbed0694cc44e914e433491b

**Comment by sahar-fehri** [comment]
I've tested this, LGTM!

**Review by sahar-fehri — APPROVED** [review]

**Comment by salimtb** [comment]
Just tested it locally , LGTM

**Review by salimtb — APPROVED** [review]

**Pull request merged**